packages/form-layout/src/vaadin-form-item.js

```diff
@@ -180,16 +251,51 @@ class FormItem extends ThemableMixin(PolymerElement) {
     return field.validate || field.checkValidity;
   }
 
-  /** @private */
+  /**
+   * A `slotchange` event handler for the label slot.
+   *
+   * - Ensures the label id is assigned to the label node if the node is provided.
+   * - Ensures the label node is linked to the first field node via the `aria-labelledby` attribute
+   * if both nodes are provided, and unlinked otherwise.
+   *
+   * @private
+   */
+  __onLabelSlotChange() {
+    if (this.__labelNode) {
+      this.__labelNode.id = this.__labelId;
+      this.__linkLabelToField(this.__fieldNode);
+    } else {
+      this.__unlinkLabelFromField(this.__fieldNode);
+    }
+  }
+
+  /**
+   * A `slotchange` event handler for the content slot.
+   *
+   * - Ensures the label node is only linked to the first field node via the `aria-labelledby` attribute.
+   * - Sets up an observer for the `required` attribute changes on the first validatable field
+   * to reflect the attribute on the component. Ensures the observer is disconnected from the field
+   * as soon as it is removed or replaced by another one.
+   *
+   * @private
+   */
   __onContentSlotChange() {
+    this.__fieldNodes.forEach((field) => {
+      if (field === this.__fieldNode && this.__labelNode) {
+        this.__linkLabelToField(field);
+      } else {
+        this.__unlinkLabelFromField(field);
```

Comment:
This unlinking doesn't work if a field is removed from the form-item, because there's no longer a reference to the discarded field. Should we rename `this.__contentField` to `this.__fieldNode` and remove the current `__fieldNode` getter?

Reply:
Refactored, please take a look.